File: app/src/main/java/com/app/catsmoker/MainActivity.java
package com.app.catsmoker;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView appInfo = findViewById(R.id.app_info);
        String info = "CatSmoker V1.5\n" +
                "Processor: " + System.getProperty("os.arch") + "\n" +
                "Model: " + Build.MODEL;
        appInfo.setText(info);

        setupButton(R.id.btn_root_lsposed, RootLSPosedActivity.class);
        setupButton(R.id.btn_shizuku, ShizukuSAFActivity.class);
        setupButton(R.id.btn_crosshair, FeaturesActivity.class);
        setupButton(R.id.btn_website, WebsiteActivity.class);
        setupButton(R.id.btn_about, AboutActivity.class);
        setupButton(R.id.btn_exit, this::finish);
    }

    private void setupButton(int buttonId, Class<?> activityClass) {
        Button button = findViewById(buttonId);
        button.setOnClickListener(v -> {
            Intent intent = new Intent(this, activityClass);
            startActivity(intent);
        });
    }

    private void setupButton(int buttonId, Runnable action) {
        Button button = findViewById(buttonId);
        button.setOnClickListener(v -> action.run());
    }
}

File: app/src/main/java/com/mb/catsmoker/MainActivity.kt
package com.mb.catsmoker

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.android.material.button.MaterialButton

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setupButtons()
    }

    private fun setupButtons() {
        findViewById<MaterialButton>(R.id.btnRootLSPosed).setOnClickListener {
            startActivitySafely(RootLSPosedActivity::class.java)
        }

        findViewById<MaterialButton>(R.id.btnShizuku).setOnClickListener {
            startActivitySafely(ShizukuActivity::class.java)
        }

        findViewById<MaterialButton>(R.id.btnCrosshair).setOnClickListener {
            startActivitySafely(CrosshairActivity::class.java)
        }

        findViewById<MaterialButton>(R.id.btnWebsite).setOnClickListener {
            openWebsite("https://catsmoker.github.io")
        }

        findViewById<MaterialButton>(R.id.btnMoreInfo).setOnClickListener {
            startActivitySafely(MoreInfoActivity::class.java)
        }

        findViewById<MaterialButton>(R.id.btnExit).setOnClickListener {
            finishAffinity()
        }
    }

    private fun startActivitySafely(cls: Class<*>) {
        try {
            startActivity(Intent(this, cls))
        } catch (e: Exception) {
            Toast.makeText(this, "Error: ${e.message}", Toast.LENGTH_SHORT).show()
        }
    }

    private fun openWebsite(url: String) {
        try {
            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(url)))
        } catch (e: Exception) {
            Toast.makeText(this, "Unable to open browser: ${e.message}", Toast.LENGTH_SHORT).show()
        }
    }
}